packages/scene-composer/src/components/panels/scene-components/AnimationComponentEditor.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useCallback, useContext } from 'react';
+import { FormField, Multiselect, SpaceBetween } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IComponentEditorProps } from '../ComponentEditor';
+import { IAnimationComponentInternal, ISceneComponentInternal, useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { OptionDefinition } from '@awsui/components-react/internal/components/option/interfaces';
+import { getClip } from '../../three-fiber/AnimationComponent/AnimationComponent';
+
+export type AnimationComponentEditorProps = IComponentEditorProps;
+
+export const AnimationComponentEditor: React.FC<AnimationComponentEditorProps> = ({
+  node,
+  component,
+}: AnimationComponentEditorProps) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const AnimationComponent = component as IAnimationComponentInternal;
```

Comment:
Why does `component` need to be casted? Can you not make `AnimationComponentEditorProps` use `IAnimationComponentInternal` as the type of `component`?